src/wmtk/operations/tri_mesh/EdgeSwapValence.cpp

```diff
@@ -0,0 +1,61 @@
+#include "EdgeSwapValence.hpp"
+#include <wmtk/SimplicialComplex.hpp>
+#include <wmtk/TriMesh.hpp>
+#include "EdgeCollapse.hpp"
+#include "EdgeSplit.hpp"
+namespace wmtk::operations::tri_mesh {
+EdgeSwapValence::EdgeSwapValence(
+    Mesh& m,
+    const Tuple& t,
+    const OperationSettings<EdgeSwapValence>& settings)
+    : EdgeSwapBase(m, t, settings.base_settings)
+    , m_settings{settings}
+{}
+
+std::string EdgeSwapValence::name() const
+{
+    return "tri_mesh_edge_swap_valence";
+}
+
+bool EdgeSwapValence::execute()
+{
+    const Tuple v0 = input_tuple();
+    const Tuple v1 = mesh().switch_vertex(input_tuple());
+    const Tuple v2 = mesh().switch_vertex(mesh().switch_edge(input_tuple()));
+    const Tuple v3 = mesh().switch_vertex(mesh().switch_edge(mesh().switch_face(input_tuple())));
+    long val0 = static_cast<long>(SimplicialComplex::vertex_one_ring(mesh(), v0).size());
```

Comment:
i think you do this long sequence in quite a few places - lets just make this a function that is shared somewhere/

Reply:
I opened #478 for that